cse-lib/network-processing/src/main/java/com/farao_community/farao/cse/network_processing/ucte_pst_change/UctePstProcessor.java

```diff
@@ -48,28 +67,17 @@ private void setTransformerInActivePowerRegulation(TwoWindingsTransformer transf
         phaseTapChanger.setTargetDeadband(5);
         phaseTapChanger.setRegulating(true);
         if (LOGGER.isInfoEnabled()) {
-            LOGGER.info(String.format("PST (%s) has been set in active power control to %.0f MW",
-                pstId, phaseTapChanger.getRegulationValue()));
+            LOGGER.info(String.format("PST at voltage level (%s) has been set in active power control to %.0f MW",
+                    voltageLevel, phaseTapChanger.getRegulationValue()));
         }
     }
 
     public void setTransformerInActivePowerRegulation(Network network) {
-        TwoWindingsTransformer transformer = network.getTwoWindingsTransformer(pstId);
-        PhaseTapChanger phaseTapChanger = getPhaseTapChanger(transformer);
-        setTransformerInActivePowerRegulation(transformer, phaseTapChanger);
-    }
-
-    public PhaseTapChanger getPhaseTapChanger(TwoWindingsTransformer transformer) {
-        if (transformer == null) {
-            throw new UctePstException(String.format(
-                "Transformer is not present in the network with the following ID : %s", pstId));
-        }
-        PhaseTapChanger phaseTapChanger = transformer.getPhaseTapChanger();
-        if (phaseTapChanger == null) {
-            throw new UctePstException(String.format(
-                "Transformer (%s) has no phase tap changer", pstId));
+        TwoWindingsTransformer transformer = getTwoWindingsTransformerAtVoltageLevelWithPhaseTapChanger(network);
+        if (transformer != null) {
+            PhaseTapChanger phaseTapChanger = transformer.getPhaseTapChanger();
+            setTransformerInActivePowerRegulation(transformer, phaseTapChanger);
```

Comment:
I think it can be good to keep the exception in case transform is null unless there is a request from the client who wants to remove it.